x-pack/plugins/alerting/server/rules_client/lib/get_alert_from_raw.ts

```diff
@@ -152,7 +152,23 @@ export function getPartialRuleFromRaw<Params extends RuleTypeParams>(
       : {}),
   };
 
-  return includeLegacyId
-    ? ({ ...rule, legacyId } as PartialRuleWithLegacyId<Params>)
-    : (rule as PartialRule<Params>);
+  // Need the `rule` object to build a URL
+  if (!excludeFromPublicApi) {
+    const viewInAppRelativeUrl =
+      ruleType.getViewInAppRelativeUrl &&
+      ruleType.getViewInAppRelativeUrl({ rule: rule as Rule<Params> });
+    if (viewInAppRelativeUrl) {
+      rule.viewInAppRelativeUrl = viewInAppRelativeUrl;
```

Comment:
nit: Should this be converted to snake case in the route handlers?

Reply:
Good catch! Fixed in https://github.com/elastic/kibana/pull/148671/commits/dd60e60fcf061f5b4a2bbf8f4fa203dca4966cdb.